simtools/db/db_handler.py

```diff
@@ -144,14 +155,15 @@ def get_model_parameters(
 
         # update using simulation model repository
         if self.mongo_db_config.get("db_simulation_model_url", None) is not None:
-            return db_from_repo_handler.update_model_parameters_from_repo(
+            _pars = db_from_repo_handler.update_model_parameters_from_repo(
                 parameters=_pars,
                 site=_site,
                 parameter_collection="telescopes",
                 telescope_name=_telescope_model_name,
                 model_version=_model_version,
                 db_simulation_model_url=self.mongo_db_config.get("db_simulation_model_url", None),
             )
+        DatabaseHandler.model_parameters[_array_elements_key] = _pars
 
         return _pars
```

Comment:
for consistency I would return `DatabaseHandler.model_parameters[_array_elements_key]` instead, although I think it really does not make a big difference.

Reply:
similar to the other functions below